### PR TITLE
Add unit tests for com.ctrip.framework.apollo.core.utils.StringUtils

### PR DESCRIPTION
**Pull request by JohnLBergqvist**
I've analysed your codebase and noticed that `com.ctrip.framework.apollo.core.utils.StringUtils` is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important.

### CODE REVIEW

File: apollo-core/src/test/java/com/ctrip/framework/apollo/core/utils/StringUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  public void testIsContainEmpty() {
+    Assert.assertFalse(StringUtils.isContainEmpty(new String[] {}));
+    Assert.assertFalse(StringUtils.isContainEmpty(new String[] {"1"}));
+    Assert.assertTrue(StringUtils.isContainEmpty(new String[] {null}));
+  }
+
+  @Test
+  public void testIsEmpty() {
+    Assert.assertFalse(StringUtils.isEmpty("1"));
+    Assert.assertTrue(StringUtils.isEmpty(null));
```
Comment (petersear) [inline apollo-core/src/test/java/com/ctrip/framework/apollo/core/utils/StringUtilsTest.java:45]: Another potential `true` test to cover all branches would be with an input of `""`

File: apollo-core/src/test/java/com/ctrip/framework/apollo/core/utils/StringUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+
+public class StringUtilsTest {
+
+  @Test
+  public void testEqualsIgnoreCase() {
```
Comment (petersear) [inline apollo-core/src/test/java/com/ctrip/framework/apollo/core/utils/StringUtilsTest.java:13]: Could add a test for two equal non-null strings.

File: apollo-core/src/test/java/com/ctrip/framework/apollo/core/utils/StringUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+  @Test
+  public void testEquals() {
+    Assert.assertFalse(StringUtils.equals(null, ""));
+    Assert.assertTrue(StringUtils.equals(null, null));
+    Assert.assertTrue(StringUtils.equals("3", "3"));
+  }
+
+  @Test
+  public void testIsBlank() {
```
Comment (petersear) [inline apollo-core/src/test/java/com/ctrip/framework/apollo/core/utils/StringUtilsTest.java:28]: Could pass in `null` for `true`

File: apollo-core/src/test/java/com/ctrip/framework/apollo/core/utils/StringUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    Assert.assertFalse(StringUtils.isBlank("\'"));
+    Assert.assertTrue(StringUtils.isBlank(""));
+  }
+
+  @Test
+  public void testIsContainEmpty() {
```
Comment (petersear) [inline apollo-core/src/test/java/com/ctrip/framework/apollo/core/utils/StringUtilsTest.java:35]: Could pass `null` for a `false` return

File: apollo-core/src/test/java/com/ctrip/framework/apollo/core/utils/StringUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+  @Test
+  public void testIsEmpty() {
+    Assert.assertFalse(StringUtils.isEmpty("1"));
+    Assert.assertTrue(StringUtils.isEmpty(null));
+  }
+
+  @Test
+  public void testIsNumeric() {
+    Assert.assertFalse(StringUtils.isNumeric("\'"));
```
Comment (petersear) [inline apollo-core/src/test/java/com/ctrip/framework/apollo/core/utils/StringUtilsTest.java:52]: Could pass in `null` for a `false` return.

File: apollo-core/src/test/java/com/ctrip/framework/apollo/core/utils/StringUtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+  @Test
+  public void testTrimToEmpty() {
+    Assert.assertEquals("1234", StringUtils.trimToEmpty("1234"));
+    Assert.assertEquals("", StringUtils.trimToEmpty(null));
+  }
+
+  @Test
+  public void trimToNull() {
+    Assert.assertEquals("foo", StringUtils.trimToNull("foo"));
```
Comment (petersear) [inline apollo-core/src/test/java/com/ctrip/framework/apollo/core/utils/StringUtilsTest.java:98]: Could pass `null` to return `null`

**Review by petersear — APPROVED** [review]
The tests look good and cover the lines.
I've added a few comments where I think branch coverage could be increased.

**Review by paulkennethkent — APPROVED** [review]
